App/Redux/TextileRedux.js

```diff
@@ -67,18 +83,28 @@ export const handleImageProgress = (state, {data}) => {
 }
 
 export const handleImageUploadComplete = (state, {data}) => {
-  const {file} = data
-  // Remove the item from Redux storage
-  const items = state.images.items.filter(item => item.remotePayloadPath !== file)
+  const { uploadId } = data
+  const existingItems = state.images.items ? state.images.items : []
+  const items = existingItems.map(item => {
+    if (item.uploadId === uploadId) {
+      return {...item, state: 'cleanup'}
+    }
+    return item
+  })
+  return state.merge({ images: { items } })
+}
+
+export const imageRemovalComplete = (state, {uploadId}) => {
+  const items = state.images.items.filter(item => item.uploadId !== uploadId)
   return state.merge({ images: { items } })
 }
 
 export const handleImageUploadError = (state, {data}) => {
-  const {file, error} = data
+  const { error, uploadId } = data
   const existingItems = state.images.items ? state.images.items : []
   const items = existingItems.map(item => {
-    if (item.remotePayloadPath === file) {
-      return {...item, state: 'error', error: error.message}
+    if (item.uploadId === uploadId) {
+      return {...item, state: 'error', error: error}
```

Comment:
I've seen that actual `Error` objects don't persist properly in Redux Persist... Better to just store `error.message` like was there before.

Reply:
oop, good catch meant to revert this